tracer/test/Datadog.Trace.Tests/SpanContextInjectorExtractorTests.cs

```diff
@@ -0,0 +1,186 @@
+        var headers = new Dictionary<string, string>();
+        var context = new SpanContext(traceId: 1, spanId: 2);
+
+        injector.InjectIncludingDsm(headers, (h, k, v) => h[k] = v, context, "Pneumatic Tube", "cashier1");
+
+        headers.Count.Should().Be(expected: 4);
+        // regular trace propagation headers
+        headers.Keys.Should().Contain("x-datadog-trace-id");
+        headers.Keys.Should().Contain("x-datadog-parent-id");
+        headers.Keys.Should().Contain("traceparent");
+        // DSM specific header
+        headers.Keys.Should().Contain("dd-pathway-ctx-base64");
+        // should not throw (i.e. should be valid base64)
+        Convert.FromBase64String(headers["dd-pathway-ctx-base64"]).Should().NotBeEmpty();
+
+        context.PathwayContext.Should().NotBeNull();
+        // I'd like to check that a checkpoint is sent, but I don't know how.
+    }
+
+    [Fact]
+    public void NoDsmInjectionIsDisabled()
+    {
+        // DSM is disabled by default
+
+        var injector = new SpanContextInjector();
+        var headers = new Dictionary<string, string>();
+        var context = new SpanContext(traceId: 1, spanId: 2);
+
+        injector.InjectIncludingDsm(headers, (h, k, v) => h[k] = v, context, "Pneumatic Tube", "cashier1");
+
+        headers.Count.Should().Be(expected: 3);
+        // regular trace propagation headers only
+        headers.Keys.Should().Contain("x-datadog-trace-id");
+        headers.Keys.Should().Contain("x-datadog-parent-id");
+        headers.Keys.Should().Contain("traceparent");
+    }
+
+    [Fact]
+    public void CanExtractAfterInjection()
+    {
+        var injector = new SpanContextInjector();
+        var headers = new Dictionary<string, string>();
+        var context = new SpanContext(traceId: 123, spanId: 456);
+
+        injector.Inject(headers, (h, k, v) => h[k] = v, context);
+
+        var extractor = new SpanContextExtractor();
+
+        var newContext = extractor.Extract(headers, (h, k) => [h[k]]);
+
+        newContext.Should().NotBeNull();
+        newContext.TraceId.Should().Be(context.TraceId);
+        newContext.Should().BeOfType(typeof(SpanContext));
+        ((SpanContext)newContext).ParentId.Should().Be(context.SpanId);
+        ((SpanContext)newContext).SamplingPriority.Should().Be(context.SamplingPriority);
+    }
+
+    [Fact]
+    public void CanExtractDsmAfterDsmInjection()
+    {
+        // enable DSM
+        Tracer.Configure(TracerSettings.Create(new Dictionary<string, object> { { ConfigurationKeys.DataStreamsMonitoring.Enabled, true } }));
+
+        var injector = new SpanContextInjector();
+        var headers = new Dictionary<string, string>();
+        var context = new SpanContext(traceId: 123, spanId: 456);
+
+        injector.Inject(headers, (h, k, v) => h[k] = v, context);
+
+        var extractor = new SpanContextExtractor();
+
+        var newContext = extractor.Extract(headers, (h, k) => [h[k]]);
```

Comment:
There is an issue here, the test is failing because the `Extract` code is expecting a lot more headers than the `Inject` code is adding, like the `x-datadog-sampling-priority` header for instance (the full list of required headers is in the next test).
I don't know if this is a bug or if I'm missing something...

Reply:
Ok, it's because I'm using a getter that throws on missing keys, and what's expected is an empty collection. I'll add that to the doc.